### PR TITLE
feat: implement modify guild emoji helper

### PR DESCRIPTION
**Pull request by ItsRqtl**
## About

This pull request is about (X), which does (Y).

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [ ] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

File: interactions/api/models/guild.py [hunk 1]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+
+        if isinstance(emoji_id, Emoji):
+            emoji = emoji_id
+        else:
+            emoji = await self.get_emoji(int(emoji_id))
+
+        _name = emoji.name if name is MISSING else name
```
Comment (Damego) [inline interactions/api/models/guild.py]: No need to send original name of emoji.
Also you could get rid of `_name` variable like this
```py
if name is not MISSING:
    payload["name"] = name
```

File: interactions/api/models/guild.py [hunk 2]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+        :return: The modified emoji object
+        :rtype: Emoji
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+
+        if isinstance(emoji_id, Emoji):
+            emoji = emoji_id
+        else:
+            emoji = await self.get_emoji(int(emoji_id))
```
Comment (Damego) [inline interactions/api/models/guild.py]: You need only ID so why you want to get emoji object?

File: interactions/api/models/guild.py [hunk 3]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+
+        _name = emoji.name if name is MISSING else name
+
+        payload: dict = {
+            "name": _name,
+        }
+
+        if roles is not MISSING:
+            _roles = [role.id if isinstance(role, Role) else role for role in roles]
```
Comment (Damego) [inline interactions/api/models/guild.py]: `_roles` variable seems like useless for me.
Also convert `Snowflake` to `int` here

File: interactions/api/models/guild.py [hunk 4]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+
+        res = await self._client.modify_guild_emoji(
+            guild_id=int(self.id), emoji_id=int(emoji.id), payload=payload, reason=reason
+        )
+
+        _emoji = Emoji(**res)
+        if self.emojis is None:
+            self.emojis = []
+        for index, item in enumerate(self.emojis):
+            if int(item.id) == int(emoji.id):
```
Comment (Damego) [inline interactions/api/models/guild.py]:
```suggestion
            if item.id == emoji.id:
```

File: interactions/api/models/guild.py [hunk 5]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
 
+    async def modify_emoji(
+        self,
+        emoji_id: Union[int, Snowflake, Emoji],
+        name: Optional[str] = MISSING,
+        roles: Optional[Union[List[Role], List[int]]] = MISSING,
+        reason: Optional[str] = None,
+    ) -> Emoji:
+        """
+        Edits an Emoji in the guild.
```
Comment (Damego) [inline interactions/api/models/guild.py]:
```suggestion
        """
        .. versionadded:: 4.4.0

        Edits an emoji in the guild.
```

**Review by Damego — CHANGES_REQUESTED** [review]

File: interactions/api/models/guild.py [hunk 6]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+
+        if roles is not MISSING:
+            _roles = [role.id if isinstance(role, Role) else role for role in roles]
+            payload["roles"] = _roles
+
+        res = await self._client.modify_guild_emoji(
+            guild_id=int(self.id), emoji_id=int(emoji.id), payload=payload, reason=reason
+        )
+
+        _emoji = Emoji(**res)
```
Comment (Damego) [inline interactions/api/models/guild.py]: Missed http client

File: interactions/api/models/guild.py [hunk 7]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+        :return: The modified emoji object
+        :rtype: Emoji
+        """
+        if not self._client:
+            raise LibraryException(code=13)
+
+        if isinstance(emoji_id, Emoji):
+            emoji_id = emoji_id.id
+        if not isinstance(emoji_id, int):
+            int(emoji_id)
```
Comment (Damego) [inline interactions/api/models/guild.py]: You could just
```py
emoji_id = int(emoji_id.id if isinstance(emoji_id, Emoji) else emoji_id)
```

File: interactions/api/models/guild.py [hunk 8]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+
+        payload: dict = {}
+
+        if name is not MISSING:
+            payload["name"] = name
+
+        if roles is not MISSING:
+            payload["roles"] = [int(role.id) if isinstance(role, Role) else role for role in roles]
```
Comment (Damego) [inline interactions/api/models/guild.py]:
```suggestion
            payload["roles"] = [int(role.id if isinstance(role, Role) else role) for role in roles]
```

File: interactions/api/models/guild.py [hunk 9]
```diff
@@ -2238,6 +2238,56 @@ async def create_emoji(
...
+
+        res = await self._client.modify_guild_emoji(
+            guild_id=int(self.id), emoji_id=emoji_id, payload=payload, reason=reason
+        )
+
+        _emoji = Emoji(**res, _client=self._client)
+        if self.emojis is None:
+            self.emojis = []
+        for index, item in enumerate(self.emojis):
+            if int(item.id) == emoji_id:
```
Comment (Damego) [inline interactions/api/models/guild.py]:
```suggestion
            if item.id == emoji_id:
```

**Comment by Damego** [comment]
Test it please

**Comment by ItsRqtl** [comment]
> Test it please

doing some work, will test it soon

**Comment by ItsRqtl** [comment]
both tested and worked

File: interactions/api/models/emoji.py [hunk 10]
```diff
@@ -109,6 +111,46 @@ async def get_all_of_guild(
         res = await client.get_all_emoji(guild_id=_guild_id)
         return [cls(**emoji, _client=client) for emoji in res]
 
+    async def modify(
+        self,
+        guild_id: Union[int, Snowflake, "Guild"],
+        name: Optional[str] = MISSING,
+        roles: Optional[Union[List[Role], List[int]]] = MISSING,
+        reason: Optional[str] = MISSING,
```
Comment (Damego) [inline interactions/api/models/emoji.py]:
```suggestion
        reason: Optional[str] = None,
```

**Review by Damego — APPROVED** [review]

**Pull request merged**